src/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -225,17 +218,14 @@ gl_sframe augment_data(const gl_sframe &data,
           // resolution to the nearest .25% interval.  Do this by multiplying by
           // 400, then do integer division by the total size, then float divide
           // by 4.0
-          int augmented_rows_completed = augmented_counter;
-          if (augmented_rows_completed % 100 == 0) {
-            d << augmented_rows_completed * 400 / total_augmented_rows / 4.0
-              << '%';
-            table.print_progress_row(augmented_rows_completed,
-                                     augmented_rows_completed, progress_time(),
-                                     d.str());
+          if (augmented_counter % 100 == 0) {
```

Comment:
`augmented_counter` is an atomic variable. You should only touch it once per iteration. Here, you are touching it up to five times. There's no guarantee it will have the same value each time.

You need to touch the value exactly once, e.g.
```
int augmented_rows_completed = augmented_counter.fetch_add(1);
```
Then use only the local copy, `augmented_rows_completed` in this iteration.

If you're unsure how to use `std::atomic` or any other language construct, please ask or read the relevant docs. Don't just guess!

Reply:
Thanks for the correction, should be fixed now.